ui/apps/platform/src/services/ReportsService.ts

```diff
@@ -129,46 +134,23 @@ export type FetchViewBasedReportHistoryServiceParams = {
     showMyHistory: boolean;
 };
 
-// @TODO: Pass API query information and set up API call to endpoint
 export function fetchViewBasedReportHistory({
     searchFilter,
     page,
     perPage,
     sortOption,
-    // @TODO: Use the showMyHistory value to determine which endpoint to use
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     showMyHistory,
 }: FetchViewBasedReportHistoryServiceParams): Promise<ViewBasedReportSnapshot[]> {
-    // @TODO: Use the params in the future API call
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const params = getListQueryParams({ searchFilter, sortOption, page, perPage });
 
-    const mockViewBasedReportJobs: ViewBasedReportSnapshot[] = [
-        {
-            reportJobId: '3dde30b0-179b-49b4-922d-0d05606c21fb',
-            isViewBased: true,
-            name: '',
-            requestName: 'SC-040925-01',
-            areaOfConcern: 'User workloads',
-            vulnReportFilters: {
-                query: 'Severity:Critical,Important+Image CVE Count:>0',
-            },
-            reportStatus: {
-                runState: 'GENERATED',
-                completedAt: '2024-11-13T18:45:32.997367670Z',
-                errorMsg: '',
-                reportRequestType: 'ON_DEMAND',
-                reportNotificationMethod: 'DOWNLOAD',
-            },
-            user: {
-                id: 'sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin',
-                name: 'admin',
-            },
-            isDownloadAvailable: true,
-        },
-    ];
+    const endpoint = showMyHistory
+        ? '/v2/reports/view-based/my-history'
+        : '/v2/reports/view-based/history';
 
-    return Promise.resolve(mockViewBasedReportJobs);
+    return axios.get<ReportHistoryResponse>(`${endpoint}?${params}`).then((response) => {
+        const snapshots = response.data?.reportSnapshots ?? [];
+        return snapshots.filter(isViewBasedReportSnapshot);
+    });
 }
```

Comment:
I noticed the need to filter based on the type predicate here. Do the `'/v2/reports/view-based/history'` and `'/v2/reports/configurations/history'` really return a mix of view-based and configuration snapshots?

Reply:
Technically, the backend protos are returning `ReportHistoryResponse` for both configured report snapshots and view-based snapshots. There can be certain fields applicable to one but not the other. The API shouldn't return a mix of report snapshots (hopefully), but I'm doing this to narrow the type to a `ViewBasedReportSnapshot` (UI specific type). It makes it easier to differentiate between them in our UI code.

I had raised a concern about using the same type/structure for both during the planning phase, but it wasn't feasible on the backend to separate them. I'm hoping we can revisit it in the future.